plugins/common/tls/config.go

```diff
@@ -206,23 +210,74 @@ func (c *ServerConfig) TLSConfig() (*tls.Config, error) {
 func makeCertPool(certFiles []string) (*x509.CertPool, error) {
 	pool := x509.NewCertPool()
 	for _, certFile := range certFiles {
-		pem, err := os.ReadFile(certFile)
+		cert, err := os.ReadFile(certFile)
 		if err != nil {
 			return nil, fmt.Errorf("could not read certificate %q: %w", certFile, err)
 		}
-		if !pool.AppendCertsFromPEM(pem) {
+		if !pool.AppendCertsFromPEM(cert) {
 			return nil, fmt.Errorf("could not parse any PEM certificates %q: %w", certFile, err)
 		}
 	}
 	return pool, nil
 }
 
-func loadCertificate(config *tls.Config, certFile, keyFile string) error {
-	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+func loadCertificate(config *tls.Config, certFile, keyFile, privateKeyPassphrase string) error {
+	var (
+		cert      tls.Certificate
+		certBytes []byte
+		keyBytes  []byte
+		err       error
+	)
+	certBytes, err = os.ReadFile(certFile)
 	if err != nil {
-		return fmt.Errorf("could not load keypair %s:%s: %w", certFile, keyFile, err)
+		return err
 	}
 
+	keyBytes, err = os.ReadFile(keyFile)
+	if err != nil {
+		return err
+	}
+
+	keyPEMBlock, _ := pem.Decode(keyBytes)
+	if keyPEMBlock == nil {
+		return errors.New("failed to decode private key: no PEM data found")
+	}
+	//nolint:staticcheck // SA1019 Disable linting for deprecated x509.IsEncryptedPEMBlock call
+	if x509.IsEncryptedPEMBlock(keyPEMBlock) {
```

Comment:
@jdstrand thoughts on the usage of this deprecated function?

Reply:
IMO we should not introduce this. https://go-review.googlesource.com/c/go/+/264159 states:
```
crypto/x509: deprecate legacy PEM encryption

It's unfortunate that we don't implement PKCS#8 encryption ([#8860](https://go.dev/issue/8860))
so we can't recommend an alternative but PEM encryption is so broken
that it's worth deprecating outright.
```

We upgrade Golang very regularly and I fear we could find ourselves in a situation where when we move forward, the API is removed/defaults changed/new compile options needed/etc. It sounds like we should be using [https://github.com/golang/go/issues/8860](PKCS#8 encryption), but it isn't implemented in the stdlib. I suggest either waiting until it is, contributing to Go upstream to implement it or using another library in the meantime.

Reply:
I understand the reservations about usage of deprecated APIs. In the absence of those functions, we can still support encrypted private keys in the PKCS#8 format which has been encrypted with any one of the PKCS#5 v2 algorithms. We would not be able to support encrypted PKCS#1 private keys. But one can always convert from PKCS#1 to encrypted PKCS#8 prior to using it with telegraf.

#### Convert from PKCS#1 to PKCS#8
```
$ openssl pkcs8 -topk8 -v2 des3 -in key.pkcs1.pem -out key.pkcs8.pem -passin pass:changeme -passout pass:changeme

$ cat key.pkcs8.pem
-----BEGIN ENCRYPTED PRIVATE KEY-----
<redacted>
-----END ENCRYPTED PRIVATE KEY-----
```
**NOTE**: Its important to use `-v2` as that is what is supported by this external library.

Reply:
Thank you @jdstrand 

@rajiv-k - will you be updating the PR to remove this section? 

As a part of this PR it would be good to add notes about this functionality to the TLS docs as well: https://github.com/influxdata/telegraf/blob/master/docs/TLS.md

Thanks!